File: majority-element/MajorityElement.js
var majorityElement = (array) => {
    let majority = 0;
    let map = new Map();
 
    for(const item of array) {
      let value = map.get(item);
      map.set(item, value === undefined ? 1: value + 1);
      lastMajority = map.get(majority) === undefined ? 0 : map.get(majority);
      currentMajority = map.get(item);
      majority = currentMajority > lastMajority ? item : majority;
    }
    return majority;
 }
 
 exports.majorityElement = majorityElement;

File: majority-element/MajorityElement.java
import java.util.*;

class MajorityElement {
  public int MajorityElement(int[] array) {
    int majority = -1;
    Map<Integer, Integer> map = new HashMap<>();
    
    for(var item: array) {
      if(!map.containsKey(item)) {
        map.put(item, 1);
      } 
      map.put(item, map.getOrDefault(item, 0) + 1);
      int current = map.get(item);
      int last = map.getOrDefault(majority, 0);
      majority = current > last ? item : majority;
    }
    return majority;
  }
}